src/orquestra/vqa/algorithms/qcbm.py

```diff
@@ -98,10 +101,40 @@ def default(
                     use_exact_expectation_values={use_exact_expectation_values}."
             )
 
-        return cls(n_qubits, n_layers, optimizer, estimation_method, n_shots)
+        return cls(
+            target_distribution,
+            n_qubits,
+            n_layers,
+            optimizer,
+            estimation_method,
+            n_shots,
+        )
+
+    def replace_target_distribution(
```

Comment:
This is the one you don't want to replace, as it defines the problem, not algorithm.
Similarly in QAOA we're not replacing Hamiltonians :)

Reply:
Done